src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -357,10 +361,22 @@ def action_version_create(
         else:
             git_policy = AllowAlwaysPolicy()
 
-        # Make sure version is not None before proceeding any further.
-        # This will raise an exception if version information is not found. Patch can be None.
         bundle_map = None
-        if not version:
+        version_resolved = None
+        patch_resolved = None
+        label_resolved = ""
+
+        # If version is specified in CLI, no version information from manifest.yml is used (except for comment, we can't control comment as of now).
+        if version is not None:
+            console.message(
+                "Version information in manifest.yml is ignored since you provided version in your command."
+            )
+            patch_resolved = patch
+            label_resolved = label if label is not None else ""
+            version_resolved = version
+
+        # When version is not set by CLI, version name is read from manifest.yml. patch and label from CLI will be used, if provided.
+        else:
```

Comment:
this method is getting really massive. How about we split the version/patch/label resolution into a helper, and return your new named tuple directly? It would greatly simplify the version creation logic proper.

Reply:
great idea. done